cache/lru_cache.py

```diff
@@ -0,0 +1,56 @@
+class LRUCacheNode:
```

Comment:
Please add brief overview of the LRU implementation. It will be easier for folks to understand the code.